pupil-spa/src/app/services/check-complete/check-complete.service.ts

```diff
@@ -0,0 +1,106 @@
+import { APP_CONFIG } from '../config/config.service';
+import { AuditService } from '../audit/audit.service';
+import { AzureQueueService } from '../azure-queue/azure-queue.service';
+import {
+  CheckSubmissionApiCalled,
+  CheckSubmissionAPIFailed,
+  CheckSubmissionAPICallSucceeded,
+} from '../audit/auditEntry';
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { SubmissionService } from '../submission/submission.service';
+import { StorageService } from '../storage/storage.service';
+import { TokenService } from '../token/token.service';
+import { queueNames } from '../azure-queue/queue-names';
+
+/**
+ * Declaration of check start service
+ */
+@Injectable()
+export class CheckCompleteService {
+
+  checkSubmissionApiErrorDelay;
+  checkSubmissionAPIErrorMaxAttempts;
+  featureUseHpa;
+  submissionPendingViewMinDisplay;
+
+  constructor(private auditService: AuditService,
+              private azureQueueService: AzureQueueService,
+              private router: Router,
+              private storageService: StorageService,
+              private submissionService: SubmissionService,
+              private tokenService: TokenService) {
+    const { featureUseHpa,
+      checkSubmissionApiErrorDelay,
+      checkSubmissionAPIErrorMaxAttempts,
+      submissionPendingViewMinDisplay,
+    } = APP_CONFIG;
+    this.checkSubmissionApiErrorDelay = checkSubmissionApiErrorDelay;
+    this.checkSubmissionAPIErrorMaxAttempts = checkSubmissionAPIErrorMaxAttempts;
+    this.featureUseHpa = featureUseHpa;
+    this.submissionPendingViewMinDisplay = submissionPendingViewMinDisplay;
+  }
+
+  /**
+   * Sleep function (milliseconds) to provide minimal display time for submission pending screen
+   * @param {Number} ms
+   * @returns {Promise.<void>}
+   */
+  private sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+  }
+
+  /**
+   * Check complete submission
+   * @param {Number} startTime Date time in milliseconds on the exact moment before check submission is called
+   * @returns {Promise.<void>}
+   */
+  public async submit(startTime): Promise<void> {
+    if (this.featureUseHpa === true) {
+      const queueName = queueNames.checkComplete;
+      const {url, token} = this.tokenService.getToken('checkComplete');
+      const retryConfig = {
+        errorDelay: this.checkSubmissionApiErrorDelay,
+        errorMaxAttempts: this.checkSubmissionAPIErrorMaxAttempts
+      };
+      this.auditService.addEntry(new CheckSubmissionApiCalled());
```

Comment:
why is this not right before the `addMessage` call?  it could potentially be confusing when reading the audit log and trying to establish what went wrong.

Reply:
@GuyHarwood the sequence is the following at the moment:
1) Audit service logs `CheckSubmissionApiCalled`
2) Payload from local storage is collected with that last audit log
3) addMessage is called

If audit service call happens right before addMessage obviously we will be missing in the payload.

Reply:
that makes sense now, thanks.